src/main/java/org/truffleruby/core/string/FrozenStrings.java

```diff
@@ -51,12 +50,12 @@ public DynamicObject getFrozenString(Rope rope) {
 
         lock.writeLock().lock();
         try {
-            string = frozenStrings.get(holder);
+            string = frozenStrings.get(rope);
             if (string == null) {
                 string = StringOperations.createFrozenString(context, rope);
                 // TODO CS 8-Jan-18 as in RopeCache the map is weak, but the key is a wrapper! So we just hold on to everything forever using this method.
```

Comment:
This comment doesn't apply any more. The key is no longer a wrapper.

The key set here should probably remain as `RopeCache` is still not correct, but the comment needs to be updated.

I think the underlying problem in both `RopeCache` and `FrozenStrings` is that it needs to be a weak-valued hash map, as well as a weak-valued one. We're doing that correctly elsewhere.

In this PR we just need to fix the comment for now though.